src/server/game/Handlers/HotfixHandler.cpp

```diff
@@ -78,6 +79,16 @@ void WorldSession::HandleHotfixRequest(WorldPackets::Hotfix::HotfixRequest& hotf
                 hotfixData.Data = boost::in_place();
                 storage->WriteRecord(hotfixData.RecordID, GetSessionDbcLocale(), *hotfixData.Data);
             }
+            else if (blobData)
```

Comment:
storage will be NULL for db2s we dont load (will crash on HasRecord)

Reply:
We'd only reach this block in case db2 hasn't that record: `if (storage->HasRecord(hotfixData.RecordID))`, never the less this else if block doesn't work properly this way. Switching the blocks should do tho. So we kinda primarily use `if (blobData)` and only use `storage->HasRecord(hotfixData.RecordID)` when there is no blobdata available.

Btw, shouldn't we differ between locales here aswell? Quite hard for some tables when just sending pure data - yes, but just thinking about this.

Reply:
No, being at this line requires `if (int32 const* hotfix = Trinity::Containers::MapGetValuePtr(hotfixes, hotfixId))` to pass. That data is loaded from hotfix_data BUT entries that point to db2s that are not loaded are SKIPPED. If you change that to be not skipped then you get a crash because `storage` is NULL